embark-consult.el

```diff
@@ -103,42 +103,100 @@
-                                   'follow-link t
-                                   'help-echo mouse-msg))
-             (nl (propertize "\n" 'occur-target loc))
-             (this-buf (marker-buffer loc)))
-          (unless (eq this-buf last-buf)
-            (insert (propertize
-                     (format "lines from buffer: %s\n" this-buf)
-                     'face list-matching-lines-buffer-name-face))
-            (setq last-buf this-buf))
-          (insert (concat lineno contents nl))))
-      (goto-char (point-min))
-      (occur-mode))
+      (save-excursion
+        (let ((mouse-msg "mouse-2: go to this occurrence")
+              last-buf)
+          (dolist (line lines)
+            (pcase-let*
+                ((`(,loc . ,num) (consult--get-location line))
+                 ;; the text properties added to the following strings are
+                 ;; taken from occur-engine
+                 (lineno (propertize (format "%7d:" num)
+                                     'occur-prefix t
+                                     ;; Allow insertion of text at the end
+                                     ;; of the prefix (for Occur Edit mode).
+                                     'front-sticky t
+                                     'rear-nonsticky t
+                                     'occur-target loc
+                                     'follow-link t
+                                     'help-echo mouse-msg))
+                 (contents (propertize (embark-consult--strip line)
+                                       'occur-target loc
+                                       'occur-match t
+                                       'follow-link t
+                                       'help-echo mouse-msg))
+                 (nl (propertize "\n" 'occur-target loc))
+                 (this-buf (marker-buffer loc)))
+              (unless (eq this-buf last-buf)
+                (insert (propertize
+                         (format "lines from buffer: %s\n" this-buf)
+                         'face list-matching-lines-buffer-name-face))
+                (setq last-buf this-buf))
+              (insert (concat lineno contents nl))))))))
+  (occur-mode)
+  (pop-to-buffer buf))
+
+(defun embark-consult-export-location-grep (lines)
+  "Create an occur or grep mode buffer listing LINES.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (let ((buf (generate-new-buffer "*Embark Export Grep*"))
+        (count 0)
+        prop)
+    (with-current-buffer buf
+      (insert (propertize "Exported line search results (file-backed buffers only):\n" 'wgrep-header t 'font-lock-face '(:weight bold)))
+      (save-excursion (let (last-buf
+                            filename
+                            (not-saved-buffers '()))
+                        (dolist (line lines)
+                          (pcase-let*
+                              ((`(,loc . ,num) (consult--get-location line))
```

Comment:
@amosbird, I appreciate your feedback. I'm using this functionality frequently myself. It would be great meaningfully extend embark further if @oantolin can look into it again. I hope he's well and sound and we'll hear from him soon.